uilib/src/components/Loader/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { Svg, Circle, G, Rect } from "react-native-svg";
+import { useTheme } from "styled-components/native";
+import { TouchableOpacity } from "react-native";
+
+type Props = {
+  // float number between 0 and 1
+  progress?: number;
+
+  // function triggered when pressing the loader
+  onPress?: () => void;
+
+  // Display the square in the middle of the loader
+  displayCancelIcon?: boolean;
+};
+
+const radius = 25;
+const strokeWidth = 2;
+const normalizedRadius = radius - strokeWidth / 2;
+const circumference = normalizedRadius * 2 * Math.PI;
+
+const ProgressLoader = ({
+  progress = 0,
+  onPress,
+  displayCancelIcon = true,
+}: Props): React.ReactElement => {
+  const { colors } = useTheme();
+  const backgroundColor = colors.palette.primary.backgroundLight;
+  const progressColor = colors.palette.primary.dark;
+
+  const strokeDashoffset = circumference - progress * circumference;
+  return (
+    <TouchableOpacity disabled={!onPress} {...{onPress} }>
```

Comment:
why this syntax ?
```jsx
  <TouchableOpacity disabled={!onPress} {...{onPress} }>
```

this is more readable i believe
```suggestion
    <TouchableOpacity disabled={!onPress} onPress={onPress}>
```

Reply:
Indeed, i was afraid of crashing if onPress was null, but if it's disabled already ... my mind was elsewhere, fixed :+1: